### DISCUSSION TITLE
Is it possible to include additional .so files in the wheel?

### DISCUSSION THREAD

**Discussion by acpreble** [PyO3/maturin#1697] [Q&A]
We have some people that want to also use the library output we're creating in C and C++ without Python bindings. We'd like to also define those functions in the project and export them, which seems to work, but the end users have to link Python against the .so to resolve missing references to all the CPython functions that are referenced. We'd love to decouple that and include the core as a separate .so file, but can't figure out if that's even possible. Is it?

Yes, we could just make that separately and crate it. A nice mashup of technological and political limitations have us looking up clever alternatives. It isn't the end of the world, and I know it's kind of silly to be distributing a C library this way. Still, I figured I'd ask and see if there was some trick. We're pretty new to the Maturin and even Rust's own Cargo stuff, so maybe there is some magic to perpetuate the bad idea. ;)

**Comment by messense** [comment]
There is an `include` option for including addition files: https://www.maturin.rs/config.html#maturin-options, which might work for you.

Even if maturin doesn't support it, you can still use [`wheel unpack`](https://wheel.readthedocs.io/en/stable/reference/wheel_unpack.html) and [`wheel pack`](https://wheel.readthedocs.io/en/stable/reference/wheel_pack.html) to do it manually.

**Reply by acpreble** [reply]
I checked that include option. I would have figured that would do it, but I just couldn't get it to include anything in the wheel. I got to the point where I had in my prproject.toml:

```
[tool.maturin]
features = ["pyo3/extension-module"]
python-source = "pysrc"
module-name="redacted_stuff_here"
include=["*.*"]
```
I figured I'd get the working directory in there (pyproject.toml, for example). I don't get anything new from when I didn't have include in the first place. It seems like a real simple option to use! Am I doing something spectacularly silly?
